### DISCUSSION TITLE
Attribs and visibility in inserts. Attsync function

### DISCUSSION THREAD

**Discussion by Gashmen** [mozman/ezdxf#653] [Q&A]
Dear @mozman ,
Discribe: I want to write my issue by easy words for my understanding. 

My issue is create function, that can be work like _.attsync * in AutoCad. I do import from DBFile by use Importer and I have new blockdatabase in new dxf file -> I create in block by use add_blockref('Block',insert = (0,0)) and I don't have attribs and visabilities -> I create attribs with the same name in DBFile  by blockref.add_attrib() -> create new attrib.(But I know, I have this attrib and after my actions in autodesk I use attsync and all attribs go on its positions with all properties(style, color...).) -> go to the CAD -> see attribs in 0,0(I know,that add_attrib's insert in default ==(0,0,0)) and block print with all visabilities. 

I need to save files with visibility so that the visibility is saved when using doc.save().When I'm using this function, library save dxf file and change something in dxf. I tried to search differences beetwen 2 files( dxf after CAD and dxf after ezdxf), but it was without results.

![Capture1](https://user-images.githubusercontent.com/90321185/158367273-f6b4310d-a58c-41d6-9c11-ab25598612b5.PNG)
![Capture2](https://user-images.githubusercontent.com/90321185/158367575-6f1bff25-315f-4b5f-a972-18b643296834.PNG)
![Capture3](https://user-images.githubusercontent.com/90321185/158370429-67d56394-8a12-4e7b-acc7-e1a9cf448133.PNG)

I create function that take my attribs from DBFile and connect to my block with all properties, but I can't do it for visibility... 

Regards,
Gashmen

**Comment by mozman** [comment]
I don't understand your question nor your problem.

**Reply by Gashmen** [reply]
I'll try to describe my problem again,sorry. 
I have a block in which several visibilitys are created. In the properties panel in AutoCAD, these visibility are in the custom section. In the example above, I created a simple box that has 2 visibility, a circle and a polyline. If I create an empty dwg file, insert this block there, save it to dxf, read doc = ezdxf.readfile, and simply resave with the doc.save function.
By opening the saved file, there will be an overlay of custom visibility on top of each other. At the same time, comparing the dxf file after saving in AutoCAD and after saving through your library, I did not see significant differences.
